cli/integration/tests/waiter/test_cli.py

```diff
@@ -0,0 +1,180 @@
+import logging
+import uuid
+
+import pytest
+
+from tests.waiter import util, cli
+
+
+@pytest.mark.cli
+@pytest.mark.timeout(util.DEFAULT_TEST_TIMEOUT_SECS)
+class WaiterCliTest(util.WaiterTest):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.waiter_url = util.retrieve_waiter_url()
+        util.init_waiter_session(cls.waiter_url)
+
+    def setUp(self):
+        self.waiter_url = type(self).waiter_url
+        self.logger = logging.getLogger(__name__)
+
+    def test_basic_create(self):
+        token_name = self.current_name()
+        util.delete_token_if_exists(self.waiter_url, token_name)
+        cp = cli.create_minimal(self.waiter_url, token_name)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        try:
+            token = util.load_token(self.waiter_url, token_name)
+            self.assertIsNotNone(token)
+            self.assertEqual('shell', token['cmd-type'])
+            resp = util.session.get(self.waiter_url, headers={'X-Waiter-Token': token_name})
+            self.assertEqual(200, resp.status_code, resp.text)
+        finally:
+            util.delete_token(self.waiter_url, token_name)
+
+    def test_failed_create(self):
+        service = util.minimal_service_description(cpus=0)
+        cp = cli.create_from_service_description(self.waiter_url, self.current_name(), service)
+        self.assertEqual(1, cp.returncode, cp.stderr)
+        self.assertEqual(b'Service description using waiter headers/token improperly configured\ncpus must be a '
+                         b'positive number..\n', cp.stderr)
```

Comment:
Can we check contains on individual fragments instead of exact equality?

Reply:
Done